### DISCUSSION TITLE
Token recognition error

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#2190] [Q&A]
Hi, I have a process as below:
```
process merging {
publishDir params.out, mode:'copy'
input:
tuple val(chrom),path(bgens) from imputed.map{it}
tuple val(chrom_mg),path(beds) from impute_merge.map{it}.first()
val flag from done_ch
output:
tuple val("${chrom}"),path("${chrom}.*") into merged
script:
def (bed,bgen,bim,fam,gz,sample,stats) = bgens
def (bed_mg,bgen_mg,bim_mg,fam_mg,gz_mg,sample_mg,stats_mg) = beds
"""
echo "${fam}" | sed "1d" | sed "s\/....\$//" > "${chrom}.merge-list.txt"
plink --bfile "${bed_mg}" "${bim_mg}" "${fam_mg}" --merge-list "${chrom}.merge-list.txt" --make-bed --out "${chrom}.step10.imputed.merged  || true
"""
}
```

I am getting error: `Script compilation error: cause: Token regnition error at: ' echo "${fam}" | sed "1d" | sed "s\/....\$//" > "${chrom}.merge-list.txt"` Details attached. Not sure what I am doing wrong. First time I am getting this error.
![token-recognition-error-nextflow](https://user-images.githubusercontent.com/12833907/120860849-9d489000-c586-11eb-8d2d-a8ffa1c1f434.png)

**Comment by pditommaso** [comment]
This looks a problem escaping the `$` in the `sed` oneliner. Better to convert it into a separate Bash script and add into the pipeline `bin/` folder https://www.nextflow.io/docs/latest/sharing.html#third-party-scripts
